### ISSUE TITLE
Terminal interception can break when used with nvm

### ISSUE THREAD

**Issue by pimterry** [httptoolkit/feedback#41]
> Does this work with NVM?
> I notice you override node, I get `usage: dirname path` and nothing is executed when I am trying to run my node application. Node/NPM commands give this.

---

_Does this affect you too? Click the +:smiley: icon in the top right and add a :+1: on issues to help decide where HTTP Toolkit goes next, or [go take a look & vote on the most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by lehtu** [comment]
Same happens here too, can't use npm nor yarn

**Comment by pimterry** [comment]
@lehtu I've been looking into this, and I'm not sure if it's directly caused by nvm (I'm actually been using nvm myself, with no problems).

Are you seeing the exact same `usage: dirname path` error as above?

The node override I'm injecting into PATH is here: https://github.com/httptoolkit/httptoolkit-server/blob/master/overrides/path/node. It:

* Uses `dirname $0` to get the name of the current folder and remove it from PATH
* Finds that path to `node` now (i.e. the path to the _real_ node)
* Resets PATH
* Runs the real node, with an extra argument prefixed, to inject some extra JS for interception

It looks like `dirname $0` is failing, seemingly because `$0` is just not defined, which is surprising - in bash that should be the name of the currently executing script.

Can you let me know:

* If you are indeed seeing the same `usage: dirname path` error as above
* What OS you're using
* What `/usr/bin/env bash --version` prints on your machine
* What `which node` prints when run inside an HTTP Toolkit terminal

**Comment by pimterry** [comment]
In fact I think I've got a fix for this. @lehtu, ignore the above - can you just run `which node` in an HTTP Toolkit terminal, and let me know if there's a space in the path printed on your machine?

It seems that in that case right now the node & php don't work correctly, and that could cause the error shown here. Fixed in https://github.com/httptoolkit/httptoolkit-server/commit/13de676b26d630584b3272a6acc97143660bdd80, it'll be in the next server release. Let me know if that's not issue though, and I'll do more investigating!

**Comment by tomsabin** [comment]
Jumping into this issue too, as I also see the `usage: dirname path` error when running some `npm` and `node` commands.

Here's the correct output from a normal terminal window (i.e. not initialised by HTTP Toolkit):

```bash
$ npm -v
6.9.0
$ node -v
v12.3.1
$ nvm --version
0.33.11
```

and the same commands on the HTTP Toolkit initialised terminal window:

```bash
$ npm -v
usage: dirname path
$ node -v
usage: dirname path
$ nvm --version
0.33.11
```

---

>  can you just run which node in an HTTP Toolkit terminal, and let me know if there's a space in the path printed on your machine?

```
$ which node
/Users/tomsabin/Applications/HTTP Toolkit.app/Contents/Resources/app/httptoolkit-server/overrides/path/node
```

You'd be right there about the space in the application name.

I renamed the application to "HTTPToolkit" and loaded the app again and can verify that I'm not longer receiving that error. Here's the same output from the HTTP Toolkit terminal:

```
$ npm -v
6.9.0
$ node -v
v12.3.1
$ nvm --version
0.33.11
$ which node
/Users/tomsabin/Applications/HTTPToolkit.app/Contents/Resources/app/httptoolkit-server/overrides/path/node
```

**Comment by pimterry** [comment]
That's perfect, thanks @tomsabin. All good then I think, I'll close this issue as soon as that fix is released.

**Comment by pimterry** [comment]
The fix for this has now been released in server 0.1.26.

If you hover over the HTTP Toolkit logo on the left of the app, it'll tell you the version you're currently using. Whilst the app is open it'll update in the background, so the next run you'll get the new version.

I'll close this now - thanks for the reports everybody!
